### ISSUE TITLE
the second params is undefined in beforeRender  function ，when use the ajax type to render data  

### ISSUE THREAD

**Issue by luckydogM** [devbridge/jQuery-Autocomplete#736]
```
var settingAutocomplete ={
            deferRequestBy: 300,   // 延时，单位毫秒
            serviceUrl:'//static.ickimg.com/data.js',
            dataType: 'jsonp',
            ajaxSettings:{
                jsonpCallback:'think'
            },
            transformResult: function(response) {
                return {
                    suggestions: $.map(response.suggestions, function(dataItem) {
                        return { value: dataItem.pro_sno, data: dataItem.about };
                    })
                };
            },
            onSelect: function(suggestion){},
            beforeRender:function(container,suggestions){
                console.log(suggestions);//print undefined
                if(suggestions.length>0){
                   for(var i=0;i<suggestions.length;i++){
                        container.find(".autocomplete-suggestion:nth-child("+(i+1)+")").append("<span>约"+suggestions[i].data+"</span>");
                         }
                    }
                }
            },
};
```

**Comment by tkirda** [comment]
This just means that suggestions were not set, or set to undefined value. Parameter is passed always, if you look at the source code.

**Comment by luckydogM** [comment]
### this place you forget a params ? so if i add a params in this function (beforeRender.call(that.element, container,that.suggestions);)  that works  and  in the custom options this  beforeRender function i can get the seconed params
`$.each(that.suggestions, function (i, suggestion) {
                if (groupBy){
                    html += formatGroup(suggestion, value, i);
                }
                html += '<div class="clearfix ' + className + '" data-index="' + i + '">' + formatResult(suggestion, value) + '</div>';
            });
            this.adjustContainerWidth();
            noSuggestionsContainer.detach();
            container.html(html);
            if ($.isFunction(beforeRender)) {
               beforeRender.call(that.element, container );
            }`

**Comment by luckydogM** [comment]
sorry,ignore it;
i think our project is not the newest overlay,
this place was changed in  you newest overlay js
